### PR TITLE
Add dynamic facets query & take care of ACL for business fields

### PR DESCRIPTION
**Pull request by nlegoff**

### CODE REVIEW

File: lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php [hunk 1]
```diff
@@ -420,6 +406,49 @@ private function createRecordQueryParams($ESQuery, SearchEngineOptions $options,
...
+            $can_see_business = (null !== $acl && $acl->can_see_business_fields($databox));
+            $rights[$databox->get_sbas_id()] = $can_see_business;
+        }
+
+        $params = [];
+        foreach ($recordHelper->getFieldsStructure() as $name => $field) {
+            if (!$field['searchable'] || !$field['to_aggregate']) {
+                continue;
+            }
+            $params[$name] = [
```
Comment (mdarse) [inline lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php]: I can see Gizeh pyramids here, maybe something like `$query['bool']['should']['...']` will be better.

File: lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php [hunk 2]
```diff
@@ -420,6 +406,49 @@ private function createRecordQueryParams($ESQuery, SearchEngineOptions $options,
...
+                'filter' => [
+                    'bool' => [
+                        'must' => [
```
Comment (mdarse) [inline lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php]: You don't need to nest `term`in a `bool`filter.

File: lib/classes/ACL.php [hunk 3]
```diff
@@ -1585,6 +1585,17 @@ public function set_limits($base_id, $limit, DateTime $limit_from = null, DateTi
         return $this;
     }
 
+    public function can_see_business_fields(\databox $databox)
+    {
+        foreach($databox->get_collections() as $collection) {
```
Comment (mdarse) [inline lib/classes/ACL.php]: Can you put a comment explaining why it depends on a single databox right.

**Comment by nlegoff** [comment]
ping @mdarse

**Pull request merged**